tools/buildsys/src/builder/error.rs

```diff
@@ -52,6 +52,18 @@ pub(crate) enum Error {
         var: String,
         source: std::env::VarError,
     },
+
+    #[snafu(display("Unknown architecture: '{}'", arch))]
+    UnknownArch {
+        arch: String,
+        source: serde_plain::Error,
+    },
+
+    #[snafu(display(
+        "Unsupported architecture '{}', check the variant's Cargo.toml for supported architectures",
```

Comment:
We could include the list of supported architectures here, since we have it in the check, and that might be more helpful / immediate...